### PR TITLE
refactor: di containers

### PR DESCRIPTION
**Pull request by bojiang**
## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Review by aarnphm — COMMENTED** [review]
FYI @bojiang @jjmachan this will break some operator right?

**Comment by bojiang** [comment]
@aarnphm I know that but thx for the reminder

**Comment by bojiang** [comment]
I think bentoctl doesn't requires DeploymentContainer anymore. https://github.com/bentoml/bentoctl/search?q=_internal
need an improvement from @jjmachan

**Comment by parano** [comment]
@bojiang could you verify the operators code are not using this?

**Comment by jjmachan** [comment]
the aws-lambda operator and azure-function is using this. I can make the change with a try - catch import statement. Would that be enought

**Comment by jjmachan** [comment]
after talking with @bojiang we are thinking of adding a bentoml_configuation.yaml file to the image created by the operator that will turn off the Prometheus. This would remove the dependency on the `_internal` API.

**Review by sauyon — APPROVED** [review]
:+1: from me.

**Review by aarnphm — APPROVED** [review]
👍🏼

File: bentoml/_internal/configuration/containers.py [hunk 1]
```diff
@@ -285,22 +289,22 @@ def access_control_options(
 
     api_server_workers = providers.Factory[int](
         lambda workers: workers or (multiprocessing.cpu_count() // 2) + 1,
-        api_server_config.workers,
+        config.bento_server.workers,
     )
-    service_port = api_server_config.port
-    service_host = api_server_config.host
+    service_port = config.bento_server.port
+    service_host = config.bento_server.host
```
Comment (parano) [inline bentoml/_internal/configuration/containers.py:295]: this should be api_server now? @sauyon @ssheng

**Review by aarnphm — APPROVED** [review]

**Comment by aarnphm** [comment]
@jjmachan let's coordinate corresponding fixes on Monday for bentoctl?

**Pull request merged**